src/savegamedump.cpp

```diff
@@ -6,17 +6,31 @@
 #include "core/GZipFormat.h"
 #include <SDL.h>
 
+int info()
+{
+	printf(
+		"savegamedump - Dump saved games to JSON for easy inspection.\n"
+		"All paths are relative to the pioneer data folder.\n"
+		"USAGE: savegamedump [--pretty] <input> [output]\n");
+	return 1;
+}
+
 extern "C" int main(int argc, char **argv)
 {
-	if (argc < 2 || argc > 3) {
-		printf(
-			"savegamedump - Dump saved games to JSON for easy inspection.\n"
-			"All paths are relative to the pioneer data folder.\n"
-			"USAGE: savegamedump <input> [output]\n");
-		return 1;
-	}
+	if (argc < 2) return info();
+
+	int indent = -1;
+	int shift = 0;
+
 	std::string filename = argv[1];
-	std::string outname = argc > 2 ? argv[2] : filename + ".json";
+	if (filename == "--pretty") {
+		indent = 2;
+		shift = 1;
+	}
```

Comment:
Note: I prefer to run dumped savegames through https://jsonviewer.stack.hu/, but this is a good secondary option if your text editor is happy with highlighting 1MB+ JSON files.

Reply:
Fortunately, I don't experience any problems with large files (in particular json) in vim. And all my bindings with me!